kite-data/kite-data-oozie/src/main/java/org/kitesdk/data/oozie/KiteURIHandler.java

```diff
@@ -0,0 +1,156 @@
+import org.slf4j.LoggerFactory;
+
+/**
+ * A Kite URI handler that works with {@link Signalable} views.
+ *
+ * To be considered as {@link #exists(URI, Configuration, String) existing} the
+ * view must have been signaled as ready.
+ */
+public class KiteURIHandler implements URIHandler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KiteURIHandler.class);
+
+  private Set<String> supportedSchemes;
+  private List<Class<?>> classesToShip;
+
+  @Override
+  public void init(final Configuration conf) {
+    supportedSchemes = new HashSet<String>();
+    final String[] schemes = conf.getStrings(
+        URIHandlerService.URI_HANDLER_SUPPORTED_SCHEMES_PREFIX
+            + this.getClass().getSimpleName()
+            + URIHandlerService.URI_HANDLER_SUPPORTED_SCHEMES_SUFFIX, "view", "dataset");
+    supportedSchemes.addAll(Arrays.asList(schemes));
+    classesToShip = new KiteLauncherURIHandler().getClassesForLauncher();
+  }
+
+  @Override
+  public Set<String> getSupportedSchemes() {
+    return supportedSchemes;
+  }
+
+  @Override
+  public Class<? extends LauncherURIHandler> getLauncherURIHandlerClass() {
+    return KiteLauncherURIHandler.class;
+  }
+
+  @Override
+  public List<Class<?>> getClassesForLauncher() {
+    return classesToShip;
+  }
+
+  @Override
+  public DependencyType getDependencyType(final URI uri)
+      throws URIHandlerException {
+    return DependencyType.PULL;
+  }
+
+  @Override
+  public void registerForNotification(final URI uri, final Configuration conf,
+      final String user, final String actionID) throws URIHandlerException {
+    throw new UnsupportedOperationException(
+        "Notifications are not supported for " + uri);
+  }
+
+  @Override
+  public boolean unregisterFromNotification(final URI uri, final String actionID) {
+    throw new UnsupportedOperationException(
+        "Notifications are not supported for " + uri);
+  }
+
+  @Override
+  public Context getContext(final URI uri, final Configuration conf,
+      final String user) throws URIHandlerException {
+    return null;
+  }
+
+  @SuppressWarnings("rawtypes")
+  @Override
+  public boolean exists(final URI uri, final Context context) throws URIHandlerException {
+    try {
+      View<GenericRecord> view = Datasets.load(uri);
+      if(view instanceof Signalable) {
+        return ((Signalable)view).isReady();
+      }
+    } catch (IllegalArgumentException ex) {
+      LOG.error("the URI {} was not a view or dataset", uri);
+    } catch (DatasetNotFoundException ex) {
+      LOG.error("the dataset for the URI {} did not actually exist", uri);
+    }
+    // didn't meet all the requirements for a URI/view that we want to use with oozie
+    return false;
+  }
+
+  @Override
+  public boolean exists(final URI uri, final Configuration conf, final String user)
+      throws URIHandlerException {
+    // currently does not handle access limitations between the oozie user and
+    // a potential dataset owner
```

Comment:
Were you thinking along the route of reading the signal files directly (so we'd be able to flex the configuration and user used for the FileSystem access)? That seems like it would bind us very directly to the current implementation, although it would work.